### DISCUSSION TITLE
Get boundary entities inside select box

### DISCUSSION THREAD

**Discussion by yojeep** [MPh-py/MPh#159] [General]
Hi!
Is it possible to realize the same functions as "mphselectbox" in matlab, or how can I  get the boundary entities's coordinates to decide which boundary to apply boundary conditions?

**Comment by john-hen** [comment]
> Is it possible to realize the same functions as "mphselectbox" in matlab

Yes. There's no reason to believe that the Matlab LiveLink (or even the Comsol GUI) has any kind of "backdoor" that you don't have access to from Python. You can use "any and all functionality offered by the [Comsol API](https://comsol.com/documentation/COMSOL_ProgrammingReferenceManual.pdf)", as the documentation explains in section ["Access the full Comsol API"](https://mph.readthedocs.io/en/1.2/demonstrations.html#access-the-full-comsol-api).

>  or how can I […]

Ultimately, you're asking how the Matlab LiveLink's `mphselectbox()` function does what it does. I, for one, do not know. And we can't look that up, it's closed-source. Maybe somebody else reads this and has the answer. If not, it is to be found *somewhere* in the Comsol documentation.

Personally, I've never set up my models in a way where I would need that functionality. If I ever wanted to select entities based on coordinates, I'd just add a "box selection" to the Comsol model, and then use it as a "named selection" wherever else needed. This also works if the geometry is parametrized, as you can refer to the same (global) parameters in the box selection's settings.

One could also add such a box selection temporarily to the model at hand, and then just read out the entity numbers it returns. Maybe that's how `mphselectbox` does it. Or the Comsol API has a dedicated function for that. Again, I've not looked into that.

**Reply by yojeep** [reply]
Thank you! I will try to add a "box selection" to do the same thing!
